### PR TITLE
FW-5082 Add Multiple Reasons to the Join Request API

### PR DESCRIPTION
**Pull request by bronson-c**
### Description of Changes
- Adds the ability to add multiple reasons to a join request (in a similar manner to the dictionary sub-models)
- The `user` field on join requests now automatically selects the user who sent the request
- Shifted reason choice values by 10 at the request of @gmcauliffe 
- Bonus: enumfield error response now prints valid options

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable
- [x] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

**Comment by bronson-c** [comment]
Adding a note here for reminder: Do not merge until after 0.3 release!!

File: firstvoices/backend/models/join_request.py [hunk 1]
```diff
@@ -17,18 +17,18 @@ class JoinRequestStatus(models.IntegerChoices):
...
+    OTHER = 10, _("Other")
+    LANGUAGE_LEARNER = 20, _("Learning the language")
+    LANGUAGE_TEACHER = 30, _("Teaching the language")
+    FLUENT_SPEAKER = 40, _("Fluent speaker")
+    LANGUAGE_INTEREST = 50, _("Interested in languages")
+    HERITAGE = 60, _("Part of my heritage")
+    COMMUNITY_MEMBER = 70, _("Member of this community/nation")
+    COMMUNITY_STAFF = 80, _("Working with this community/nation")
+    RECONCILIATION = 90, _("Reconciliation")
+    FV_TEAM = 100, _("Part of this FirstVoices Language Team")
```
Comment (sarahfirstvoices) [inline firstvoices/backend/models/join_request.py:31]: @gmcauliffe Do you need an API to retrieve this list, or are you ok with making a copy of it in the frontend?

File: firstvoices/backend/serializers/join_request_serializers.py [hunk 2]
```diff
@@ -11,32 +12,66 @@
...
+        fields = ("reason",)
+        model = JoinRequestReason
 
 
 class JoinRequestDetailSerializer(WritableSiteContentSerializer):
     url = SiteHyperlinkedIdentityField(
         view_name="api:joinrequest-detail", read_only=True
     )
-    user = UserLookupField(required=True, allow_null=False)
+    user = UserDetailSerializer(required=False, allow_null=False, read_only=True)
```
Comment (sarahfirstvoices) [inline firstvoices/backend/serializers/join_request_serializers.py:30]: Minor but I think if read_only is True you don't need to set required=False here

File: firstvoices/backend/models/join_request.py (hunk 1 above)
Comment (gmcauliffe) [reply firstvoices/backend/models/join_request.py:31]: I'm ok with making a copy in the FE. Thanks :)

**Review by sarahfirstvoices — APPROVED** [review]
Love these changes!

**Pull request merged**